src/mam4xx/aging.hpp

```diff
@@ -74,73 +73,81 @@ void mam_pcarbon_aging_frac(
   const Real soa_vol =
-      aero_species(iaer_soa).molecular_weight / aero_species(iaer_soa).density;
-  const Real bc_vol =
-      aero_species(iaer_bc).molecular_weight / aero_species(iaer_bc).density;
+      _molecular_weight_soa * 1000.0 / aero_species(iaer_soa).density;
+  const Real bc_vol = aero_species(iaer_bc).molecular_weight * 1000.0 /
+                      aero_species(iaer_bc).density;
   const Real pom_vol =
-      aero_species(iaer_pom).molecular_weight / aero_species(iaer_pom).density;
-  const Real mom_vol =
-      aero_species(iaer_mom).molecular_weight / aero_species(iaer_mom).density;
+      _molecular_weight_pom * 1000.0 / aero_species(iaer_pom).density;
+  const Real mom_vol = aero_species(iaer_mom).molecular_weight * 1000.0 /
+                       aero_species(iaer_mom).density;
 
-  const Real fac_m2v_eqvhyg_aer =
-      soa_vol * aero_species(iaer_soa).hygroscopicity;
+  // (Bad Constants) for hygroscopicitiy
+  constexpr Real hygro_soa = 0.14000000000000001;
+  constexpr Real hygro_so4 = 0.50700000000000001;
+  const Real fac_m2v_eqvhyg_aer = soa_vol * hygro_soa / hygro_so4;
 
-  // for default MAM4 only so4 and soa contribute to aging
-  const Real vol_shell = qaer_cur[iaer_so4][nsrc] * so4_vol +
-                         qaer_cur[iaer_soa][nsrc] * fac_m2v_eqvhyg_aer;
+  Real qaer_del_cond_tmp =
+      qaer_del_cond[iaer_so4][imom_pc] * so4_vol +
+      qaer_del_cond[iaer_soa][imom_pc] * fac_m2v_eqvhyg_aer;
 
-  const Real qaer_del_cond_tmp =
-      haero::max(qaer_del_cond[iaer_so4][nsrc] * so4_vol +
-                     qaer_del_cond[iaer_soa][nsrc] * fac_m2v_eqvhyg_aer,
-                 std::numeric_limits<float>::epsilon());
   const Real qaer_del_coag_tmp =
       qaer_del_coag_in[iaer_so4][ipair] * so4_vol +
       qaer_del_coag_in[iaer_soa][ipair] * fac_m2v_eqvhyg_aer;
 
-  frac_cond = 1.0 - qaer_del_cond_tmp / (qaer_del_cond_tmp +
-                                         haero::max(qaer_del_coag_tmp, 0.0));
+  qaer_del_cond_tmp = haero::max(qaer_del_cond_tmp, 1e-35);
+
+  frac_cond = qaer_del_cond_tmp /
+              (qaer_del_cond_tmp + haero::max(qaer_del_coag_tmp, 0.0));
+
   frac_coag = 1.0 - frac_cond;
 
+  // for default MAM4 only so4 and soa contribute to aging
+  const Real vol_shell = qaer_cur[iaer_so4][imom_pc] * so4_vol +
+                         qaer_cur[iaer_soa][imom_pc] * fac_m2v_eqvhyg_aer;
+
   const int spec_modes[3] = {iaer_bc, iaer_pom, iaer_mom};
   const Real core_volumes[3] = {bc_vol, pom_vol, mom_vol};
   Real vol_core = 0.0;
   for (int mi = 0; mi < 3; ++mi) {
     const int ispec = spec_modes[mi];
-    vol_core += qaer_cur[ispec][nsrc] * core_volumes[mi];
+    vol_core += qaer_cur[ispec][imom_pc] * core_volumes[mi];
   }
 
   const Real fac_volsfc = haero::exp(
-      2.5 * haero::log(haero::pow(mam4::modes(nsrc).mean_std_dev, 2.0)));
+      2.5 * haero::pow(haero::log(mam4::modes(imom_pc).mean_std_dev), 2.0));
```

Comment:
you could replace this `pow` expression with `haero::square(haero::log(mam4::modes(imom_pc).mean_std_dev))` if you wanted.

Reply:
Done.